src/chttpd/src/chttpd_httpd_handlers.erl

```diff
@@ -328,8 +338,14 @@ handler_info('GET', [Db, <<"_design">>, Name | Path], _) ->
         'attachment.name' => filename:join(Path)
     }};
 
-handler_info('PUT', [Db, <<"_design">>, Name | Path], _) ->
-    {'db.design.doc.attachment.write', #{
+handler_info('PUT', [Db, <<"_design">>, Name | Path], Req) ->
+    ActionName = case chttpd:qs_value(Req, "rev") of
```

Comment:
So I thought about just checking the rev like this, but the problem here is that for PUT and POST, the rev can be embed'ed in the request body and is not guaranteed to be in the query string. I looked at `chttpd:qs_value` and that appears to just be looking at the query string params, so I don't think this is currently sufficient to properly tag these as write vs update.

Reply:
I think `_bulk_docs` may have similar issues, where you can both add new docs and update existing docs in the same request, and would presumably need to inspect each element of the `docs` field to know what's actually going on?

Reply:
@chewbranca updated

Reply:
@jaydoane This is getting into expensive call territory. It would force us to traverse docs multiple times.

Reply:
@iilyak I think it's just another example of `_bulk_docs` being a lousy API, and I agree it's expensive. However, each request in principle can do a lot of work, so the upfront cost _might_ be worth it for correctness? Dunno, tbh.